src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentHelper.cs

```diff
@@ -229,7 +238,7 @@ private ViewComponentDescriptor SelectComponent(Type componentType)
                 throw new ArgumentNullException(nameof(descriptor));
             }
 
-            var context = new ViewComponentContext(descriptor, arguments, _viewContext, writer);
+            var context = new ViewComponentContext(descriptor, arguments, _htmlEncoder, _viewContext, writer);
```

Comment:
Should we make the encoder part of the `ViewContext` instead? It seems to show up everywhere that's used.